src/language-js/embed.js

```diff
@@ -25,12 +25,18 @@ const placeholderPiecesToStringArray = pieces =>
   pieces.map(({ isPlaceholder, string, placeholder }) =>
     isPlaceholder ? placeholder : string
   );
-const removeCSSComments = string =>
+
+const removeCSSCommentsAndQuotedString = string =>
   string
+    // remove quoted strings
+    .replace(/(['"]).*?\1/, "$1$1")
```

Comment:
We also need to handle escapes in strings. This regex should do it:

```
(['"])(?:(?!\2)[^\\\r\n\f]|\\(?:\r\n|[\s\S]))*(\2)
```

Copied from https://github.com/lydell/css-tokens

We also need to handle unquoted URLs, such as:

```
background-image: url(https://example.com/foo.png);
```

Regex:

```
url\((?!\s*["'])\s*(?:[^"'()\\\s]|\\(?:[\da-fA-F]{1,6}\s?|[^\r\n\f]))*\s*\)
```

Reply:
I understand to not break css code, escapes need to be handled, but this function only need make sure strings after `//`(not a comment) is not lost, I even strip all strings inside quotes. 

and I already add test for escaped quote

`url()` is hot handled

Reply:
**Prettier pr-6794**
[Playground link](https://deploy-preview-6794--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAucBnVADAOlAhgAmACNcwBrAcwCcIBXKAEwFoBLAW1wriX1qoBsAFAAsYMAA6okAemlwAHrjbj+cAHSQ20gGYQIa8VAoBKANwASYLgC+1nBhAAaEBHEwW0VMlC4qNAO4ACr4IXii4AG4QLAxOIERUpGRwMADK4qQsRsgwVLRwzqJs-ADqwizwqBlgcKmhFSwRFQCeyGhezlmocFQwgYkUHMjauPzdzgBWqPIAQonkKalKcAAyWXDDo+MgU-KpWRSqAIq0EPCbYwUgGVTdVG0kRHD8ceJUWTAlMTDCyAAcAAZnG8IN0SolxG03nA7hENs4AI6neD9VxhEC4VBMKBwOAMPFxKhwJEsIn9ThDJAjS7ObpsFg5PJXVAHY7IjZUrZXGC4IhfBg-ZAAJmcuVwLH4BwAwhA2JSQFBoPCQLRugAVXlhanbCL5ACSjAQaTA7zcAEFGKkYM1VBdurYgA)
```sh
--parser babel
```

**Input:**
```jsx
css`
a {background-image: url(https://example.com/foo.png);${a}}
`
```

**Output:**
```jsx
SyntaxError: (postcss) CssSyntaxError Unknown word (2:55)
  1 | 
> 2 | a {background-image: url(https://example.com/foo.png);pprreettttiieerrfgfyruzjaaaaaaaaaaaaaaaarreeiitttteerrpp}
    |                                                       ^
  3 | 
```

Reply:
Hmm, I guess I need to dive deeper into how this PR actually works…

Reply:
this part is clean before/after to check if `templateLiteral` is a `orphan placeholder`, beause `postcss` can't parse

```css
a {
  placeholder
}
```

I need fake it into
```css
a {
  prop: placeholder;
}
```

I clean before/after comments, and check is this `templateLiteral`

```js
           (!after ||
              endsWithLineBreak ||
              after[0] === ";" ||
              after[0] === "}") &&
            (!before ||
              before.slice(-1) === ";" ||
              before.slice(-1) === "{" ||
              before.slice(-1) === "}")
```

Reply:
seem I don't even need handle `url(//example.com)` because you can't wrote this in jsx,

you need escape `url(\/\/example.com)`